server/svix-server/tests/e2e_attempt.rs

```diff
@@ -145,3 +150,130 @@ async fn test_list_attempts_by_endpoint() {
     receiver_1.jh.abort();
     receiver_2.jh.abort();
 }
+
+async fn test_attempt_retries(endpoint: &str, expected_status: i16) {
+    let _ = dotenv::dotenv();
+    let mut cfg = svix_server::cfg::load().unwrap().as_ref().clone();
+    cfg.queue_type = svix_server::cfg::QueueType::Memory;
+    cfg.retry_schedule = (0..2)
+        .into_iter()
+        .map(|_| Duration::from_millis(1))
+        .collect();
+
+    let (client, _jh) = start_svix_server_with_cfg(&cfg);
+
+    let app_id = create_test_app(&client, "app1").await.unwrap().id;
+
+    let endp_id = create_test_endpoint(&client, &app_id, endpoint)
+        .await
+        .unwrap()
+        .id;
+
+    let msg = create_test_message(&client, &app_id, serde_json::json!({"test": "data1"}))
+        .await
+        .unwrap();
+
+    for i in &cfg.retry_schedule {
+        sleep(*i).await;
+    }
+
+    run_with_retries(|| async {
+        let list: ListResponse<MessageAttemptOut> = client
+            .get(
+                &format!("api/v1/app/{}/attempt/msg/{}", app_id, &msg.id),
+                StatusCode::OK,
+            )
+            .await
+            .unwrap();
+
+        if list.data.len() != cfg.retry_schedule.len() + 1 {
+            anyhow::bail!("Attempt count does not match retry_schedule length");
+        }
+        for i in list.data.iter() {
+            if i.status != MessageStatus::Fail {
+                anyhow::bail!("Unexpected MessageStatus");
+            }
+            if i.response_status_code != expected_status {
+                anyhow::bail!(
+                    "HTTP status should be 200, found {}",
+                    i.response_status_code
+                );
+            }
+            if i.endpoint_id != endp_id {
+                anyhow::bail!("Unexpected endpoint id");
+            }
+        }
+        Ok(())
+    })
+    .await
+    .unwrap();
+}
+
+#[tokio::test]
+async fn test_attempt_failures_are_retried() {
+    let receiver = TestReceiver::start(axum::http::StatusCode::FORBIDDEN);
+
+    // HTTP non-200 failures:
+    test_attempt_retries(&receiver.endpoint, 403).await;
+    receiver.jh.abort();
+
+    // non-HTTP-related failures:
+    test_attempt_retries(&receiver.endpoint, 0).await;
+}
+
+#[tokio::test]
+async fn test_successful_attempt() {
```

Comment:
Isn't this exactly almost the same as `test_attempt_retries` but with a 200 response? To be honest, I'm not 100% what's being tested in this test. Just a normal delivery path, no? If so, I'd rather have a different file for this that just tests e2e message delivery.

Reply:
It is indeed just checking for a succesful attempt (ie w/ no retries). I have refactored to remove the mostly-redundant code. This scenario is not previously tested as far as I can tell, so leaving it in.